File: my_crypt.js
const crypto = require('crypto');
const base64 = require('base64-js');

function encryptMessage(messageJson, key) {
  const keyBytes = base64.toByteArray(key);
  const cipher = crypto.createCipheriv('aes-256-ecb', keyBytes, null);
  const messageBuffer = Buffer.from(messageJson, 'utf-8');
  const encrypted = Buffer.concat([cipher.update(messageBuffer), cipher.final()]);
  return base64.fromByteArray(encrypted);
}

function decryptMessage(encryptedMessageBase64, key) {
  const keyBytes = base64.toByteArray(key);
  const encryptedMessageBytes = base64.toByteArray(encryptedMessageBase64);
  try {
    const decipher = crypto.createDecipheriv('aes-256-ecb', keyBytes, null);
    const decrypted = Buffer.concat([decipher.update(encryptedMessageBytes), decipher.final()]);
    return JSON.parse(decrypted.toString('utf-8'));
  } catch (e) {
    console.debug(`Decryption failed, error: ${e}`);
    return null;
  }
}

function generateKey() {
  const keyBytes = crypto.randomBytes(32);
  return base64.fromByteArray(keyBytes);
}

File: my_crypt.py
import base64
import json
import logging
import secrets

from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad


# logger = logging.getLogger('myapp')

def encrypt_message(message_json, key):
    key = base64.b64decode(key)

    cipher = AES.new(key, AES.MODE_ECB)
    ciphertext = cipher.encrypt(pad(message_json.encode('utf-8'), AES.block_size))

    # 将密文转换为字符串
    encrypted_message = base64.b64encode(ciphertext).decode('utf-8')
    return encrypted_message

def decrypt_message(encrypted_message_base64, key):
    key = base64.b64decode(key)
    try:
        # 将密文字符串转换为字节串
        ciphertext = base64.b64decode(encrypted_message_base64.encode('utf-8'))

        # 解密
        cipher = AES.new(key, AES.MODE_ECB)
        decrypted_message = unpad(cipher.decrypt(ciphertext), AES.block_size)

        return json.loads( decrypted_message.decode('utf-8'))
    except Exception as e:
        # logger.debug(f"解密失败，错误原因：{e}")
        return None
    
    
def generate_key():
    # 生成一个32字节（256位）的随机密钥
    key_bytes = secrets.token_bytes(32)
    
    # 将密钥编码为Base64字符串
    key_str = base64.b64encode(key_bytes).decode('utf-8')
    return key_str

